matchtree.go

```diff
@@ -866,17 +873,41 @@ func (d *indexData) newMatchTree(q query.Q) (matchTree, error) {
 			return &noMatchTree{"const"}, nil
 		}
 	case *query.Language:
-		code, ok := d.metaData.LanguageMap[s.Language]
-		if !ok {
-			return &noMatchTree{"lang"}, nil
+		code, codeOk := d.metaData.LanguageMap[s.Language]
+		var mt matchTree
+
+		if !codeOk {
+			mt = &noMatchTree{"lang"}
+		} else {
+			mt = &docMatchTree{
+				reason:  "language:" + s.Language,
+				numDocs: d.numDocs(),
+				predicate: func(docID uint32) bool {
+					return d.languages[docID] == code
+				},
+			}
 		}
-		return &docMatchTree{
-			reason:  "language",
-			numDocs: d.numDocs(),
-			predicate: func(docID uint32) bool {
-				return d.languages[docID] == code
-			},
-		}, nil
+
+		extsForLang := enry_data.ExtensionsByLanguage[s.Language]
```

Comment:
So what does this provide versus just using LanguageMap like the old code did? ie if we always have Enry setting the doc.Language can't we just rely on that?

Reply:
Yes, but we can't rely on that due to the 256 language limit and because we still need to support non-enry-identified shards in a backwards compatible manner.